File: LeapYearChecker.js
function calculate() {
    var input = document.getElementById("yearinput").value
    if (input % 4 == 0) {
        if (input % 100 == 0) {
            if (input % 400 == 0) {
                document.getElementById("result").innerHTML = "You are lucky! This is a leap year!"
            } else {
                document.getElementById("result").innerHTML = "Bad Luck! This is not a leap year!"
            }
        } else {
            document.getElementById("result").innerHTML = "You are lucky! This is a leap year!"
        }
    } else {
        document.getElementById("result").innerHTML = "Bad Luck! This is not a leap year!"
    }
}

File: LeapYearChecker.py
thisyear = int(input("Input this year's year date(eg: 2022): "))
if thisyear % 4 == 0:
    if thisyear % 100 == 0:
        if thisyear % 400 == 0:
            print("You are lucky! This is a leap year!")
        else:
            print("Bad Luck! This is not a leap year.")
    else:
        print("You are lucky! This is a leap year!")
else:
    print("Bad Luck! This is not a leap year.")